pkg/env/env.go

```diff
@@ -382,13 +382,18 @@ func (e *testEnv) Run(m *testing.M) (exitCode int) {
 		// context passed down to each setup
 		if ctx, err = setup.run(ctx, e.cfg); err != nil {
 			klog.Errorf("%s failure: %s", setup.role, err)
+			exitCode = 1
 			break
 		}
 	}
 	e.ctx = ctx
 
-	// Execute the test suite
-	return m.Run()
+	// Execute the test suite (if the setup was successful)
+	if err == nil {
+		exitCode = m.Run()
+	}
+
+	return exitCode
```

Comment:
Why not just `return 1` from line 385 instead ?

Reply:
Thanks for your feedback. I thought it would be easier to read this way, since the context has to be assigned to the environment anyway: `e.ctx = ctx`.

I can change it to:
```
	if ctx, err = setup.run(ctx, e.cfg); err != nil {
			klog.Errorf("%s failure: %s", setup.role, err)
			e.ctx = ctx
			return 1
	}
	e.ctx = ctx
```

What do you think?

Reply:
amd why you assign the ctx to `e.ctx`

Reply:
ah this is actually not necessary, I thought `e.ctx` is used for the `Finish` actions, but it is the local copy, i.e., `ctx`.